src/math/fermats_little_theorem.rs

```diff
@@ -0,0 +1,134 @@
+use rand::Rng;
+
+/// Performs modular exponentiation.
+///
+/// This function computes `(base ^ exp) mod modulus` efficiently using the method
+/// of exponentiation by squaring.
+///
+/// # Parameters
+///
+/// - `base`: The base value (u64).
+/// - `exp`: The exponent (u64).
+/// - `modulus`: The modulus (u64).
+///
+/// # Returns
+///
+/// The result of `(base ^ exp) mod modulus`.
+fn mod_exp(base: u64, exp: u64, modulus: u64) -> u64 {
+    let mut result = 1;
+    let mut base = base % modulus;
+    let mut exp = exp;
+
+    while exp > 0 {
+        if exp % 2 == 1 {
+            result = (result * base) % modulus;
+        }
+        exp /= 2;
+        base = (base * base) % modulus;
+    }
+
+    result
+}
```

Comment:
Why not to use [`modular_exponential`](https://github.com/TheAlgorithms/Rust/blob/0dbaff560587087363a704694895bf310d269197/src/math/modular_exponential.rs#L60)?